Tests/ReceiptParserTests/ReceiptParserFetchingTests.swift

```diff
@@ -17,4 +17,117 @@ import XCTest
 
 class ReceiptParserFetchingTests: XCTestCase {
 
+    private let parser: PurchasesReceiptParser = .default
+    private var mockFileReader: MockFileReader!
+    private var mockBundle: MockBundle!
+
+    override func setUp() {
+        super.setUp()
+
+        self.mockFileReader = .init()
+        self.mockBundle = .init()
+    }
+
+    func testParseWithNoReceiptThrowsError() throws {
+        self.mockBundle.receiptURLResult = .sandboxReceipt
+
+        do {
+            _ = try self.fetchAndParse()
+            fail("Expected error")
+        } catch PurchasesReceiptParser.Error.failedToLoadLocalReceipt {
+            // expected error
+        } catch {
+            fail("Unexpected error: \(error)")
+        }
+
+        expect(self.mockFileReader.invokedContentsOfURL).to(haveCount(1))
+        expect(self.mockFileReader.invokedContentsOfURL.first?.value) == 1
+    }
+
+    func testParseWithNilURLThrowsError() throws {
+        self.mockBundle.receiptURLResult = .nilURL
+
+        do {
+            _ = try self.fetchAndParse()
+            fail("Expected error")
+        } catch {
+            expect(error).to(matchError(PurchasesReceiptParser.Error.receiptNotPresent))
```

Comment:
I noticed here we are not using the same system as in the other 2 tests, where we catch the specific test we expect, and then a final catch to throw an error. Is there a reason for that?

Reply:
+1, this seems like a nicer way to check for the error being the right type

Reply:
This approach produces a better error when it fails and it's one line instead of 2 different catches.

In the other test we're checking with a more specific error with an associated value which we wouldn't be able to do with `matchError`.

Reply:
ohhh the associated value, I missed that. What about `testParseWithNoReceiptThrowsError`?

Reply:
Oh yeah that can be simplified. Thanks!

Reply:
Oh I see now, that one has an associated value too: `case failedToLoadLocalReceipt(Swift.Error)`